server/src/main/java/org/tctalent/server/service/db/impl/JobOppIntakeServiceImpl.java

```diff
@@ -63,7 +63,7 @@ public void update(long id, JobIntakeData data) throws NoSuchObjectException {
         jobOppIntakeRepository.save(joi);
     }
 
-    private void populateIntakeData(JobOppIntake joi, JobIntakeData data) {
+    private void populateIntakeData(JobOppIntake joi, IJobIntakeData data) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.

Reply:
Yes I saw that and I see why. It is basically counting a large number of if statements which would normally be a good indicator of complexity. But in this case it is very obvious. We only update fields when there are non null values supplied. The code is in fact trivially simple.